apps/cli-daemon/src/app/workspace/workspace.service.ts

```diff
@@ -16,73 +18,143 @@ import {
 
 import { SessionService } from '../session/session.service';
 
+import { ProjectDto, UpdateProjectDto } from './dto';
 import {
   ANGULAR_WORKSPACE_NOT_FOUND_EXCEPTION,
   BAD_PATH_EXCEPTION,
   NOT_ANGULAR_WORKSPACE_EXCEPTION,
 } from './entities';
 
-export const ng = 'npx ng new';
+const ANGULAR_JSON = '/angular.json';
 
 @Injectable()
 export class WorkspaceService {
   private readonly logger = new Logger(WorkspaceService.name);
-  readonly ng = ng;
+
   constructor(private sessionService: SessionService) {}
 
-  async readWorkspace(path: string): Promise<WorkspaceDefinition> {
+  async isAngularWorkspace(path: string): Promise<boolean> {
     try {
-      return (
-        await devKitReadWorkspace(
-          path,
-          createWorkspaceHost(new NodeJsSyncHost())
-        )
-      ).workspace;
+      return !!(await devKitReadWorkspace(
+        path,
+        createWorkspaceHost(new NodeJsSyncHost())
+      ));
+    } catch {
+      return false;
+    }
+  }
+
+  async connect(path: string): Promise<void> {
+    try {
+      await devKitReadWorkspace(
+        path,
+        createWorkspaceHost(new NodeJsSyncHost())
+      );
+      this.sessionService.setCwd(path);
       // eslint-disable-next-line @typescript-eslint/no-explicit-any
     } catch (err: any) {
-      const errorMessage = err['message'];
-      switch (errorMessage) {
-        case BAD_PATH_EXCEPTION:
-          throw new BadRequestException(BAD_PATH_EXCEPTION);
-        case NOT_ANGULAR_WORKSPACE_EXCEPTION:
-          throw new NotFoundException(ANGULAR_WORKSPACE_NOT_FOUND_EXCEPTION);
-        default:
-          this.logger.error(errorMessage);
-          throw new InternalServerErrorException(err);
-      }
+      throw this.handleReadWorkspaceException(err);
     }
   }
 
   async readWorkspaceProjectNames(): Promise<string[]> {
-    const workspace = await this.readWorkspace(this.sessionService.cwd);
+    const workspaceDefinition = await this._readWorkspaceDefinition();
 
-    return Array.from(workspace.projects.keys());
+    return Array.from(workspaceDefinition.projects.keys());
   }
 
-  async readWorkspaceProject(
-    name: string
-  ): Promise<ProjectDefinition | undefined> {
-    const workspace = await this.readWorkspace(this.sessionService.cwd);
+  async readWorkspaceProject(name: string): Promise<ProjectDto> {
+    return new ProjectDto(await this._readProjectDefinition(name));
+  }
+
+  async updateWorkspaceProject(
+    name: string,
+    updateProjectDto: UpdateProjectDto
+  ): Promise<void> {
```

Comment:
don't you think update should return the updated result?

Reply:
right, I planned to do it and forgot, thanks